habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ContainerizeDepsMojo.java

```diff
@@ -241,6 +259,12 @@ protected void collectHabushuDependenciesAsProjects(MavenProject currentProject,
             if (habushuDeps.contains(toGav(project))) {
                 logger.info("Found project {} as habushu-type dependency.", project);
                 habushuProjects.add(project);
+                if (this.singleHabushuDependency != null) {
+                    if (toGav(project).equals(toGav(this.singleHabushuDependency))) {
+                        String path = project.getBasedir().getPath();
+                        this.singleHabushuDependencyBaseDir = path.substring(path.indexOf(this.anchorSourceDirectory.getName()));
```

Comment:
A: I feel like this might be easier to follow if we have `getHabushuProjects` return both the full set of projects and the primary project in one result object, then have `copySourceCode` return the relevant path of the primary project.